src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -31,6 +31,12 @@ async function boot(options?: Partial<WebAssemblyStartOptions>): Promise<void> {
   };
 
   // Configure navigation via JS Interop
+  // Wrap getBaseURI \ getLocationHref to return unmarshaled values
+  const getBaseUri = window['Blazor']._internal.navigationManager.getBaseURI;
+  const getLocationHref = window['Blazor']._internal.navigationManager.getLocationHref;
+  window['Blazor']._internal.navigationManager.getBaseURI = () => BINDING.js_string_to_mono_string(getBaseUri());
+  window['Blazor']._internal.navigationManager.getLocationHref = () => BINDING.js_string_to_mono_string(getLocationHref());
```

Comment:
I know this will work, but it's a bit alarming that we'd be monkeypatching our own code. Makes it super hard to reason about what actually happens at runtime when reading other parts of the code.

Instead of wrapping and overwriting code like this, could we do one of the following?

 * Have some abstraction (interface, etc) around a service that supplies location info, with different implementations for Blazor Server and Blazor WebAssembly
 * Or maybe leave `Blazor._internal.navigationManager.getBaseURI`/`getLocationHref` alone, and add alongside them two new functions (`getBaseURIUnmarshalled`/`getLocationHrefUnmarshalled`) for the WebAssembly case only, which wrap the existing ones inside `BINDING.js_string_to_mono_string(...)`. Presumably the .NET call site is different for Server/WebAssembly, since one of them already expects marshalled interop and the other expects unmarshalled.

I guess with my second suggestion the end result is not much different from what you have here, but it's nice not to be monkeypatching :)

Reply:
> Or maybe leave Blazor._internal.navigationManager.getBaseURI/getLocationHref alone, and add alongside them two new functions (getBaseURIUnmarshalled/getLocationHrefUnmarshalled) for the WebAssembly case only, which wrap the existing ones inside BINDING.js_string_to_mono_string(...). 

That sounds reasonable. I'll do that.